packages/klevu-core/src/store/filterManager.ts

```diff
@@ -339,50 +339,31 @@ export class FilterManager {
           (subOption) => subOption.selected === true
         )
         if (selected.length === 0) {
+          params.delete(`o_${filter.key}`)
           continue
         }
         params.set(`o_${filter.key}`, selected.map((s) => s.value).join(","))
       } else if (FilterManager.isKlevuFilterResultSlider(filter)) {
         if (!filter.start || !filter.end) {
+          params.delete(`s_${filter.key}`)
           continue
         }
         params.set(`s_${filter.key}`, `${filter.start}-${filter.end}`)
       }
     }
-    /**
-     * This code will convert the URLSearchParams to a string and also remove
-     * the options that are no longer selected.
-     */
-    let paramsAsString = ""
-    for (const [key, value] of params.entries()) {
-      let append = false
-      if (key.startsWith("o_")) {
-        const optionKey = key.substring(2)
-        const filter = this.filters.find((f) => f.key === optionKey)
-        if (filter) {
-          if (
-            FilterManager.isKlevuFilterResultOptions(filter) ||
-            FilterManager.isKlevuFilterResultRating(filter)
-          ) {
-            const option = filter.options.find((subOption) =>
-              value.split(",").find((v) => v === subOption.value)
-            )
-            if (option?.selected) {
-              append = true
-            }
-          }
+
+    // delete the ones that are not in the filters
+    for (const [key] of params.entries()) {
+      if (key.startsWith("o_") || key.startsWith("s_")) {
+        const filterKey = key.substring(2)
+        const filter = this.filters.find((f) => f.key === filterKey)
+        if (!filter) {
+          params.delete(key)
         }
-      } else {
-        append = true
-      }
-      if (append) {
-        paramsAsString += `${
-          paramsAsString ? "&" : ""
-        }${key}=${encodeURIComponent(value)}`
       }
     }
 
-    return paramsAsString
+    return params.toString()
```

Comment:
this doesnt work, the url looks like this
http://localhost:3333/category?o_designer=7+for+all+mankind
or
http://localhost:3007/?o_lenses_colour=Black+-+Black
I was manually preparing the string to solve this issue.

although search passes correct value, :confused: ?